src/detector.js

```diff
@@ -174,3 +189,17 @@ function matchPartsAgainstList(source, list) {
     return target.every((part, index) => source[index] === part)
   })
 }
+
+function matchStringAgainstList(needle, list) {
+  const listMap = new Map(list);
+  return listMap.has(needle) ? needle : null;
+}
+
+function ipfsCidRegex(startEndMatch = true) {
+  // regex from https://stackoverflow.com/a/67176726
+  let reg = "Qm[1-9A-HJ-NP-Za-km-z]{44,}|b[A-Za-z2-7]{58,}|B[A-Z2-7]{58,}|z[1-9A-HJ-NP-Za-km-z]{48,}|F[0-9A-F]{50,}";
+  if(startEndMatch) {
```

Comment:
Since we always use `ipfsCidRegex` with `false` could we remove this extra complexity or does it have a use elsewhere?

Reply:
It's used with `true` on the tests https://github.com/MetaMask/eth-phishing-detect/pull/58432/files#diff-b7a738f35866f9728b7c1f8d2cb769aaaf35e55c270a530a28c3e0ec7227bd60R75